### DISCUSSION TITLE
ESP32/ILI9488 display not clearing

### DISCUSSION THREAD

**Discussion by beng2k** [Bodmer/TFT_eSPI#2450] [Q&A - General]
Hi,

I am using this code to create a smooth arc and display a value

tft.drawSmoothArc(399, 250, gaugeRadius, gaugeInnerRadius, gaugeStart, DATA6gaugeEnd, TFT_ORANGE, TFT_BLACK);
ofr.setCursor(399,215);
ofr.cprintf("%3.0f", DATA6value[0]);

Everything works but does not clear the smooth arc or numbers.
I am using OpenFontRender for the text.
The smooth arc moves up but does not clear when the values are lower.
Ive tried this using sprites and not using sprites to the same effect.

I have set background and text colours earlier in the code but the smooth arc also includes the background colour each time with no changes.

**Comment by Bodmer** [comment]
See Arc_Meter_Demo sketch.

You have to draw redraw the end of the arc with the background. To do this keep the last drawn angle. See code here:

https://github.com/Bodmer/TFT_eSPI/blob/master/examples/Smooth%20Graphics/Arc_meter_demo/Arc_meter_demo.ino#L199-L206
